src/main/kotlin/io/openfuture/state/service/WalletService.kt

```diff
@@ -0,0 +1,11 @@
+package io.openfuture.state.service
+
+import io.openfuture.state.controller.domain.dto.WalletDto
+import io.openfuture.state.controller.domain.request.SaveWalletRequest
+
+interface WalletService {
+
+    suspend fun save(request: SaveWalletRequest): WalletDto
```

Comment:
The only controller operates with DTO, Request, and Response. 
Service never care about that.

So, here should be the following signature:

```
suspend fun save(address: String, webhook: String): Wallet
```

Reply:
Sometimes, when there are a lot of arguments, it makes sense to pass the whole request object (and it is not prohibited):

```
suspend fun save(request: SaveWalletRequest): Wallet
```

However, service never response with DTO. And having a small number of parameters is clearer than passing FE Request.